CMakeLists.txt

```diff
@@ -0,0 +1,98 @@
+cmake_minimum_required(VERSION 3.15)
+
+file(STRINGS "VERSION" pVersion)
+
+project(
+  ufs_util
+  VERSION ${pVersion}
+  LANGUAGES C Fortran)
+# LANGUAGES C CXX Fortran)
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/")
+
+if(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|MinSizeRel)$")
+  message(STATUS "Setting build type to 'Release' as none was specified.")
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release"
+                                               "MinSizeRel" "RelWithDebInfo")
+endif()
+
+if(NOT CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel|GNU|Clang|AppleClang)$")
+  message(WARNING "Compiler not officially supported: ${CMAKE_Fortran_COMPILER_ID}")
+endif()
+
+if(NOT CMAKE_C_COMPILER_ID MATCHES "^(Intel|GNU)$")
+  message(WARNING "Compiler not officially supported: ${CMAKE_C_COMPILER_ID}")
+endif()
+
+find_package(PNG REQUIRED)
+find_package(ZLIB REQUIRED)
+find_package(Jasper REQUIRED)
+
+find_package(NetCDF REQUIRED C Fortran)
+#unset(CMAKE_C_COMPILER_LOADED)
+#unset(CMAKE_CXX_COMPILER_LOADED)
+find_package(MPI REQUIRED )
+find_package(ESMF MODULE REQUIRED)
+find_package(WGRIB2 REQUIRED)
+
+find_package(OpenMP REQUIRED COMPONENTS Fortran)
```

Comment:
Shouldn't threading be optional?

Reply:
If making threads optional is a best practice/industry standard, I will make this change.

Reply:
You can make it ON default, but there should be an option to not compile with threads.

Reply:
I encountered that here https://github.com/GeorgeGayno-NOAA/UFS_UTILS/issues/8

The call to `omp_get_max_threads()` needs to be conditional

Reply:
I took a closer look at the global_chgres code.  Making threads optional in that code will require more work than it is worth.   Since global_chgres is nearly obsolete (replaced by chgres_cube), could the build process be modified to compile global_chgres only if the user turns on threads?  No one outside of EMC will ever use global_chgres.

Reply:
I think that sounds fine. Just add an if around the add_subdirectory to global_chgres? @aerorahul?

Reply:
```
if (OpenMP_FOUND)
  add_subdirectory(global_chgres.fd)
endif()
```

Reply:
Great.  I will add the logic from @aerorahul.

Reply:
The threaded routines in global_chgres are also used by global_cycle.  So updating global_cycle to be thread optional will be more work than it is worth right now.  So logic was added to not build global_chgres **and** global_cycle if the user turns off threads.  No one outside EMC is using global_cycle currently and it will not be part of the next public release.